content/integrations/codecov-integration.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Codecov integration
+description: How to integrate your workflows with BrowserStack using codemagic.yaml
+weight: 7
+---
+
+**Codecov** is a dedicated code analysis tool and one of the leading code coverage solutions for mobile applications. **Codecov** can be used as part of the Codemagic CI/CD pipeline for code coverage. 
+
+## Collecting test results
+
+After writing tests with your test suite you can generate a coverage report using **lcov** and upload that coverage report to **Codecov** directly via **codemagic.yaml**. It is also possible to exit the script if code coverage is less or more than expected rates. Refer to the sample script below:
+
+```
+- name: Coverage report
+  script: |
+     HOMEBREW_NO_AUTO_UPDATE=1 brew install lcov
+     mkdir -p test-results 
+     flutter test --coverage --machine > test-results/flutter.json  
+     code_coverage=$(lcov --list $FCI_BUILD_DIR/coverage/lcov.info | sed -n "s/.*Total:|\(.*\)%.*/\1/p")
+     echo "Code Coverage: ${code_coverage}% "
+     if (( $(echo "$code_coverage < $CODE_COVERAGE_TARGET" | bc) )); then { echo "code coverage is less than expected" && exit 1; }; fi  
+  test_report: test-results/flutter.json
+```
+
+In order to store test results in a location after collecting them, just include the **test_report** field with a glob pattern matching the test result file location.
+
+
+## Create a Codecov account
+
+In order to get a dedicate Codecov token, signing up is required. You can sign up for free [here](https://about.codecov.io/): 
+
+## Configuring Codecov Code Coverage Upload:
+
+Code coverages can be submitted to the Codecov environment through Codemagic using a **cURL** request. In order to configure it correctly, codecov token is required and it can be found under the Codecov account after signing up. After getting the necessary token, it can be assigned to an enviornment variable. Environment variables can be added in the Codemagic web app using the ‘Environment variables’ tab. You can then and import your variable groups into your codemagic.yaml. For example, if you named your variable group ‘browserstack_credentials’, you would import it as follows:
```

Comment:
```suggestion
Code coverages can be submitted to the Codecov environment through Codemagic using a **cURL** request. In order to configure it correctly, a codecov token is required which can be found in your Codecov account after signing up. After getting the necessary token, it can be assigned to an environment variable. Environment variables can be added in the Codemagic web app using the ‘Environment variables’ tab. You can then and import your variable groups into your codemagic.yaml. For example, if you named your variable group ‘codecov_credentials’, you would import it as follows:
```